app/HMS/Repositories/Doctrine/DoctrineProfileRepository.php

```diff
@@ -139,6 +140,25 @@ public function findOneByDiscordUserSnowflake(string $discordUserSnowflake)
         return parent::findOneByDiscordUserSnowflake($discordUserSnowflake);
     }
 
+    /**
+     * @param Carbon join date
+     *
+     * @return Profile|null
+     */
+    public function findByJoinedOn(Carbon $joinDate)
+    {
+        $q = parent::createQueryBuilder('profile');
+
+        $q->where('profile.joinDate >= :join_date_lower')
+          ->andWhere('profile.joinDate < :join_date_upper');
+
+        $q = $q->setParameter('join_date_lower', $joinDate)
+               ->setParameter('join_date_upper', $joinDate->copy()->addDays(1));
```

Comment:
if we add `'DATE' => DoctrineExtensions\Query\Mysql\Date::class,` to `doctrine.custom_datetime_functions` config

we should be able to just do 

```suggestion
        $q->where('DATE(profile.joinDate) = :join_date');

        $q = $q->setParameter('join_date', $joinDate->toDateString());
```